lms/tasks/email_digests.py

```diff
@@ -95,24 +86,20 @@ def send_instructor_email_digest_tasks(self, *, batch_size, h_userids=None):
                             "updated_before": updated_before,
                         },
                     )
-                except Exception as exc:  # pylint:disable=broad-exception-caught
-                    LOG.exception(exc)
-                    report_exception(exc)
-                    failed_h_userids.extend(batch)
-
-            if failed_h_userids:
-                self.retry(
-                    kwargs={
-                        **self.request.kwargs,
-                        "h_userids": failed_h_userids,
-                    },
-                    countdown=_retry_countdown(self.request),
+                except Exception as err:  # pylint:disable=broad-exception-caught
+                    failed = True
+                    LOG.exception(err)
+                    report_exception(err)
```

Comment:
Here the task is looping over the list of batches and scheduling an instance of the `send_instructor_email_digests()` below for each batch. It does need to catch any exception raised when scheduling an instance of the other task so that it can continue to the next iteration of the `for` loop and an exception when scheduling one batch doesn't cause the following batches to all be skipped.